### PR TITLE
Update how versionwarning.js handles doc versions with more than one digit

### PR DESCRIPTION
**Pull request by AlenkaF**
This PR corrects the check for version number  of the documentation when adding version warnings to the html files.
See: https://github.com/apache/arrow/issues/34023

Also, additional copying of the versionwarning.js file seems to happen with the release. Currently there are three versions of the same file:

- docs/_static/versionwarning.js
- docs/dev/_static/versionwarning.js
- docs/10.0/_static/versionwarning.js

Second and third should be redundant and are removed with this PR. I will create a separate issue on the apache/arrow repo to research and correct how the versionwarning.js file is being copied.

### CODE REVIEW

**Comment by AlenkaF** [comment]
The issue about multiple copies of `versionwarning.js` in `apache/arrow-site`: https://github.com/apache/arrow/issues/34160

File: docs/_static/versionwarning.js [hunk 1]
```diff
@@ -30,6 +30,7 @@
             // Path of the page
             var location_array = location.pathname.split('/');
             var versionPath = location_array[2];
+            var versionNumber = Number (versionPath.match(/^\d+/))
```
Comment (kou) [inline docs/_static/versionwarning.js:33]:
```suggestion
            var versionNumber = Number(versionPath.match(/^\d+/))
```

**Review by kou — COMMENTED** [review]
I think that the upstream of `versionwarning.js` is https://github.com/apache/arrow/blob/master/docs/source/_static/versionwarning.js .
Why do we update `versionwarning.js` in apache/arrow-site instead of apache/arrow?
I think that this change will be overwritten in the next release.

**Comment by AlenkaF** [comment]
> I think that the upstream of `versionwarning.js` is https://github.com/apache/arrow/blob/master/docs/source/_static/versionwarning.js . Why do we update `versionwarning.js` in apache/arrow-site instead of apache/arrow? I think that this change will be overwritten in the next release.

Correct, I plan to update the file in the main repo as a part of https://github.com/apache/arrow/issues/34023. But first I would like to correct the current state on the arrow site, if possible. Plus, the multiple versions of the .js file bothers me also.

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

**Pull request merged**